### ISSUE TITLE
libroom.rir_builder error: "The rir array is too small for the maximum time recorded"

### ISSUE THREAD

**Issue by mmjamm** [LCAV/pyroomacoustics#353]
Hi!

The new RIR builder in libroom C++ extension is great - my scripts run much faster!
I did run into a regressions though.
Sometimes I'm getting the following error:
`Traceback (most recent call last):
  File "/media/data/miheikki/git/ramen/pra_test.py", line 96, in <module>
    room.compute_rir()
  File "/home/miheikki/.local/share/virtualenvs/ramen-plSAbj_6/lib/python3.10/site-packages/pyroomacoustics/room.py", line 2327, in compute_rir
    libroom.rir_builder(
RuntimeError: the rir array is too small for the maximum time recorded`

I'm able to reproduce the bug with the following code:
```
import numpy as np
import pyroomacoustics as pra

room_dims = np.array([10., 10., 10.])
room = pra.ShoeBox(
        room_dims,
        fs=24000,
        materials=None,
        max_order=22,
        use_rand_ism=False
)

source = np.array([[6.35551912],[4.33308523], [3.69586303]])
room.add_source(source)

mic_array_in_room = np.array(
    [
        [1.5205189, 1.49366285, 1.73302404, 1.67847898],
        [4.68430529, 4.76250254, 4.67956424, 4.60702604],
        [2.68214263, 2.7980202,  2.55341851, 2.72701718]
    ]
)
room.add_microphone_array(mic_array_in_room)

room.compute_rir()
```

I'm running `Python 3.10.14`
My Python env:
```
Package         Version
--------------- -------
Cython          3.0.10
numpy           2.0.0
pip             24.0
pybind11        2.12.0
pyroomacoustics 0.7.4
scipy           1.13.1
setuptools      69.5.1
wheel           0.43.0
```

I'm not able to reproduce on 0.7.3 and I haven't seen the issue prior to 0.7.4.

Cheers!

**Comment by fakufaku** [comment]
Thanks for reporting this bug! I have found the cause to be due to rounding of the delays before computing the RIR.
PR #354 fixes the bug.

**Comment by fakufaku** [comment]
Fixed in [v0.7.5](https://pypi.org/project/pyroomacoustics/0.7.5/)

**Comment by stet-stet** [comment]
Hi, I'd like to add that I encountered this on v0.7.5 today. It only occurred after simulating 10000s of files, where many of the room params and rt60 were set at random (were not logged either), so for the moment I have not been able to come up with a minimal reproduction. If I am able to reproduce this again I will report back.

**Comment by fakufaku** [comment]
@stet-stet thanks for letting me know. Happy to help debug this if you can find the room parameters that cause this.

**Comment by stet-stet** [comment]
@fakufaku I found one!
```
import numpy as np
from scipy.io import wavfile
import pyroomacoustics as pra

# The desired reverberation time and dimensions of the room
rt60_tgt = 0.451734045124395  # seconds
room_dim = [
            2.496315595944846,
            2.2147285947364708,
            3.749472153652182
        ]  # meters

sample_wav = "(anything should work)"
fs, audio = wavfile.read(sample_wav)

e_absorption, max_order = pra.inverse_sabine(rt60_tgt, room_dim)

room = pra.ShoeBox(
    room_dim, fs=fs, materials=pra.Material(e_absorption), max_order=max_order
)

room.add_source([
                0.24784311631630576,
                1.690743273038349,
                1.9570721698068267
            ], signal=audio, delay=0.) # this didn't need to be 0 either

mic_array = np.array([
            [
                0.46378325918698565,
            ],
            [
                1.5657207092343373,
            ],
            [
                3.015697444447528,
            ]
        ])

room.add_microphone_array(mic_array)

room.simulate()
```

Hope you could help debug. Thanks!

**Comment by MordehayM** [comment]
i have encountered the same problem after producing 8k examples

**Comment by fakufaku** [comment]
@MordehayM , if you have the parameters of the room that fails handy, it would be great. I would add them as a test for the fix.

**Comment by MordehayM** [comment]
I am afraid I don't have them since all the parameters were randomly sampled.

**Comment by MordehayM** [comment]
I have found one:
Room dimensions: [4.57977238 5.39054892 2.82767573], RT60: 0.5824602516207891, Absorption coefficient: 0.18257819716723472, Speaker position: [1.22812426 1.2966769  1.43330033], Mic position: [1.8020194  0.76576269 0.53980759], Max order: 83

**Comment by fakufaku** [comment]
Thank you so much, this is very useful!

**Comment by fakufaku** [comment]
@MordehayM 
I have tested with the provided parameters, but I can't reproduce the error. What was the sampling frequency?

**Comment by MordehayM** [comment]
16kHz

**Comment by fakufaku** [comment]
@MordehayM Thanks!

The issue should be fixed by #360 in [v0.7.6](https://github.com/LCAV/pyroomacoustics/releases/tag/v0.7.6).

Please let me know if there are more issues.
